packages/amplify-graphql-relational-transformer/src/resolver/ddb-references-generator.ts

```diff
@@ -0,0 +1,360 @@
+import { ObjectTypeDefinitionNode } from 'graphql';
+import {
+  DynamoDBMappingTemplate,
+  Expression,
+  ObjectNode,
+  and,
+  bool,
+  compoundExpression,
+  equals,
+  ifElse,
+  iff,
+  int,
+  isNullOrEmpty,
+  list,
+  methodCall,
+  not,
+  nul,
+  obj,
+  or,
+  print,
+  qref,
+  raw,
+  ref,
+  set,
+  str,
+  toJson,
+} from 'graphql-mapping-template';
+import {
+  ModelResourceIDs,
+  NONE_VALUE,
+  ResolverResourceIDs,
+  applyCompositeKeyConditionExpression,
+  applyKeyConditionExpression,
+  attributeTypeFromScalar,
+  setArgs,
+  toCamelCase,
+} from 'graphql-transformer-common';
+import { condenseRangeKey } from '../resolvers';
+import { BelongsToDirectiveConfiguration, HasManyDirectiveConfiguration, HasOneDirectiveConfiguration } from '../types';
+import { RelationalResolverGenerator } from './generator';
+
+const SORT_KEY_VALUE = 'sortKeyValue';
+const CONNECTION_STACK = 'ConnectionStack';
+const authFilter = ref('ctx.stash.authFilter');
+const PARTITION_KEY_VALUE = 'partitionKeyValue';
+
+export class DDBRelationalReferencesResolverGenerator extends RelationalResolverGenerator {
+  makeExpression = (keySchema: any[], connectionAttributes: string[]): ObjectNode => {
+    if (keySchema[1] && connectionAttributes[1]) {
+      let condensedSortKeyValue;
+
+      if (connectionAttributes.length > 2) {
+        const rangeKeyFields = connectionAttributes.slice(1);
+
+        condensedSortKeyValue = rangeKeyFields
+          .map((keyField, idx) => `\${${SORT_KEY_VALUE}${idx}}`)
+          .join(ModelResourceIDs.ModelCompositeKeySeparator());
+      }
+
+      return obj({
+        expression: str('#partitionKey = :partitionKey AND #sortKey = :sortKey'),
+        expressionNames: obj({
+          '#partitionKey': str(keySchema[0].attributeName),
+          '#sortKey': str(keySchema[1].attributeName),
+        }),
+        expressionValues: obj({
+          ':partitionKey': ref(`util.dynamodb.toDynamoDB($${PARTITION_KEY_VALUE})`),
+          ':sortKey': ref(`util.dynamodb.toDynamoDB(${condensedSortKeyValue ? `"${condensedSortKeyValue}"` : `$${SORT_KEY_VALUE}0`})`),
+        }),
+      });
+    }
+
+    return obj({
+      expression: str('#partitionKey = :partitionKey'),
+      expressionNames: obj({
+        '#partitionKey': str(keySchema[0].attributeName),
+      }),
+      expressionValues: obj({
+        ':partitionKey': ref(`util.dynamodb.toDynamoDB($${PARTITION_KEY_VALUE})`),
+      }),
+    });
+  };
+
+  /**
+   * Create a resolver that queries an item in DynamoDB.
+   * @param config The connection directive configuration.
+   * @param ctx The transformer context provider.
+   */
+  makeHasManyGetItemsConnectionWithKeyResolver = (config: HasManyDirectiveConfiguration, ctx: TransformerContextProvider): void => {
+    const { connectionFields, field, indexName, limit, object, references, relatedType } = config;
+    const connectionAttributes: string[] = references.length > 0 ? references : connectionFields;
```

Comment:
You're correct, this shouldn't be here because we're not implicitly creating fields in the new `references` world. Thanks for calling that out, addressed in ea6f7bd6a959b75831c03c0c71aad54d454c62f7